brew_view/metrics.py

```diff
@@ -0,0 +1,109 @@
+
+# Summaries:
+http_api_latency_total = Summary(
+    'bg_http_api_latency_seconds',
+    'Total number of seconds each API endpoint is taking to respond.',
+    ['method', 'route', 'status']
+)
+plugin_command_latency = Summary(
+    'bg_plugin_command_latency_seconds',
+    'Total time taken for a command to complete in seconds.',
+    ['system', 'instance_name', 'system_version', 'command', 'status'],
+)
+
+# Counters:
+completed_request_counter = Counter(
+    'bg_completed_requests_total',
+    'Number of completed requests.',
+    ['system', 'instance_name', 'system_version', 'command', 'status'],
+)
+request_counter_total = Counter(
+    'bg_requests_total',
+    'Number of requests.',
+    ['system', 'instance_name', 'system_version', 'command'],
+)
+
+# Gauges:
+queued_request_gauge = Gauge(
+    'bg_queued_requests',
+    'Number of requests waiting to be processed.',
+    ['system', 'instance_name', 'system_version'],
+)
+in_progress_request_gauge = Gauge(
+    'bg_in_progress_requests',
+    'Number of requests IN_PROGRESS',
+    ['system', 'instance_name', 'system_version'],
+)
+
+
+def initialize_counts():
+    for request in Request.objects(status='CREATED'):
+        queued_request_gauge.labels(
+            system=request.system,
+            system_version=request.system_version,
+            instance_name=request.instance_name,
+        ).inc()
+
+    for request in Request.objects(status='IN_PROGRESS'):
+        in_progress_request_gauge.labels(
+            system=request.system,
+            system_version=request.system_version,
+            instance_name=request.instance_name,
+        ).inc()
+
+
+def request_created(request):
+    queued_request_gauge.labels(
+        system=request.system,
+        system_version=request.system_version,
+        instance_name=request.instance_name,
+    ).inc()
+    request_counter_total.labels(
+        system=request.system,
+        system_version=request.system_version,
+        instance_name=request.instance_name,
+        command=request.command,
+    ).inc()
+
+
+def request_updated(request, status_before):
+    """Update metrics associated with a Request update
+
+    This call should happen after the save to the database.
+
+    """
+    if (
+            status_before == request.status or
+            status_before in BrewtilsRequest.COMPLETED_STATUSES
+    ):
+        return
+
+    labels = {
+        'system': request.system,
+        'system_version': request.system_version,
+        'instance_name': request.instance_name,
+    }
+
+    if status_before == 'CREATED':
+        queued_request_gauge.labels(**labels).dec()
+    elif status_before == 'IN_PROGRESS':
+        in_progress_request_gauge.labels(**labels).dec()
+
+    if request.status == 'IN_PROGRESS':
+        in_progress_request_gauge.labels(**labels).inc()
+
+    elif request.status in BrewtilsRequest.COMPLETED_STATUSES:
+        # We don't use _measure_latency here because the request times are
```

Comment:
This comment makes a little less sense now that this is not in the controllers. I like moving everything to a metrics module (as that is what we will have to do in the near future anyway). If so, it might make sense to move `_measure_latency` here too (and expose it publicly). Since there is a refactor around the corner, I'm not sure how much this matters, but it's my two cents.

Reply:
Oh wait. Nevermind, if we've moved metrics out to its own module, there should definitely be tests for this :)

Reply:
Yeah, I thought about that, but figured I'd err on the side of not changing things. I'll pull it into `metrics`.